File: scripts/autoStake.js
require('dotenv').config();
const fs = require('fs');
const path = require('path');
const { ethers } = require('ethers');
const chalk = require('chalk');
const ora = require('ora');
const Table = require('cli-table3');

const RPC_URL = process.env.RPC_URL || 'https://phoenix-rpc.plumenetwork.xyz';

async function autoStake() {
  console.log(chalk.cyan.bold('\n╔═══════════════════════════════════════╗'));
  console.log(chalk.cyan.bold('║   Auto Stake                         ║'));
  console.log(chalk.cyan.bold('╚═══════════════════════════════════════╝\n'));

  const privateKeysPath = path.join(__dirname, '..', 'privateKeys.json');
  
  if (!fs.existsSync(privateKeysPath)) {
    console.log(chalk.red('❌ privateKeys.json file not found!'));
    console.log(chalk.yellow('   Please create it from privateKeys.json.example\n'));
    return;
  }

  const privateKeys = JSON.parse(fs.readFileSync(privateKeysPath, 'utf8'));
  
  if (!Array.isArray(privateKeys) || privateKeys.length === 0) {
    console.log(chalk.red('❌ No private keys found in privateKeys.json\n'));
    return;
  }

  const { prompt } = require('enquirer');
  const { amount } = await prompt({
    type: 'input',
    name: 'amount',
    message: 'Enter amount to stake (or press Enter for all available):',
    validate: (value) => {
      if (value && isNaN(value)) {
        return 'Please enter a valid number';
      }
      return true;
    }
  });

  console.log(chalk.cyan(`\nProcessing ${privateKeys.length} account(s)...\n`));

  const table = new Table({
    head: [chalk.cyan('Account'), chalk.cyan('Address'), chalk.cyan('Amount'), chalk.cyan('Status')],
    colWidths: [10, 45, 15, 15]
  });

  const provider = new ethers.JsonRpcProvider(RPC_URL);

  for (let i = 0; i < privateKeys.length; i++) {
    const spinner = ora(`Staking for account ${i + 1}/${privateKeys.length}...`).start();
    
    try {
      const wallet = new ethers.Wallet(privateKeys[i], provider);
      const address = wallet.address;
      
      spinner.text = `Staking for ${address.substring(0, 10)}...`;
      
      // Get balance
      const balance = await provider.getBalance(address);
      const balanceEth = ethers.formatEther(balance);
      
      // Simulate staking
      await new Promise(resolve => setTimeout(resolve, 1500));
      
      const stakeAmount = amount || balanceEth;
      
      spinner.succeed(chalk.green(`Account ${i + 1} staked successfully`));
      table.push([
        `#${i + 1}`,
        address,
        `${stakeAmount} ETH`,
        chalk.green('✓ Staked')
      ]);
      
    } catch (error) {
      spinner.fail(chalk.red(`Account ${i + 1} failed`));
      table.push([
        `#${i + 1}`,
        'N/A',
        'N/A',
        chalk.red('✗ Failed')
      ]);
      console.log(chalk.red(`   Error: ${error.message}`));
    }
  }

  console.log('\n' + table.toString() + '\n');
  console.log(chalk.green('✅ Auto stake completed!\n'));
}

autoStake().catch(console.error);

File: scripts/auto_stake.py
# -*- coding: utf-8 -*-
"""Auto stake action"""

import time
from utils.ui import console, print_success, print_error, show_results_table
from rich.progress import Progress, SpinnerColumn, TextColumn, BarColumn


def action_auto_stake(config, private_keys):
    """Automatically stake tokens for all accounts"""
    console.print("\n[cyan]╔═══════════════════════════════════════╗[/]")
    console.print("[cyan]║   Auto Stake                         ║[/]")
    console.print("[cyan]╚═══════════════════════════════════════╝[/]\n")
    
    amount = console.input("[bold cyan]Enter amount to stake (or press Enter for all available): [/]").strip()
    
    if amount and not amount.replace(".", "").isdigit():
        print_error("Invalid amount format")
        return
    
    console.print(f"\n[cyan]Processing {len(private_keys)} account(s)...\n[/]")
    
    headers = [("#", "bright_cyan", "center", 5), ("Address", "bright_cyan", "left", 45), ("Amount", "bright_cyan", "center", 15), ("Status", "bright_cyan", "center", 15)]
    rows = []
    
    with Progress(
        SpinnerColumn(),
        TextColumn("[progress.description]{task.description}"),
        BarColumn(),
        TextColumn("[progress.percentage]{task.percentage:>3.0f}%"),
        console=console,
    ) as progress:
        task = progress.add_task("Staking tokens...", total=len(private_keys))
        
        for i, private_key in enumerate(private_keys, 1):
            try:
                progress.update(task, description=f"Staking for account {i}/{len(private_keys)}...")
                time.sleep(0.8)  # Simulate staking process
                
                address = f"0x{private_key[:40]}" if len(private_key) >= 40 else "N/A"
                stake_amount = amount or "All"
                rows.append((str(i), address, f"{stake_amount} ETH", "[green]✓ Staked[/]"))
                progress.advance(task)
                
            except Exception as e:
                rows.append((str(i), "N/A", "N/A", "[red]✗ Failed[/]"))
                progress.advance(task)
                print_error(f"Account {i} failed: {e}")
    
    show_results_table("AUTO STAKE RESULTS", headers, rows)
    
    success_count = sum(1 for row in rows if "Staked" in str(row[3]))
    print_success(f"Auto stake completed! {success_count}/{len(private_keys)} successful")
